### ISSUE TITLE
Manual char string search pattern

### ISSUE THREAD

**Issue by kornelski** [rust-lang/rust-clippy#12490]
### What it does

Uses of `Pattern` (in methods like `str.split()`, `str.replace()`, `str.trim_matches()`) where the pattern is a closure matching `char`s:

```rust
str.split(|c: char| c == '\n' || c == 'X') // or
str.split(|c: char| matches!(c, '\n' | 'X'))
```

can be expressed more succinctly as:

```rust
str.split(['\n', 'X'])
```

and a single-char comparison can be simplified further.

### Advantage

It makes the code much shorter, and the character set much easier to read. 

They all have about the same performance, except a single-char pattern is much faster than the other forms.

### Example

```rust
sentence.trim_end_matches(|c: char| c == '.' || c == '!' || c == '?')
```

Could be written as:

```rust
sentence.trim_end_matches(['.', '!', '?'])
```

**Comment by AurelienFT** [comment]
Hello @Vrajs16,
Are you still working on this ?
I would love to start contributing to clippy and this is an issue that interest me :)

**Comment by Vrajs16** [comment]
@AurelienFT you can claim it

**Comment by y21** [comment]
Closing as this was implemented in #12849
